DockWin.ahk

```diff
@@ -0,0 +1,207 @@
+;DockWin v0.5 - Save and Restore window positions when docking/undocking (using hotkeys) over all Windows 10 Desktops
+; Paul Troiano, 6/2014
+; Updated by Ashley Dawson 7/2015
+; Updated by Robert Niebsch 10/2016
+;
+; Hotkeys: Capslock + 0 to save; Capslock + 9 to load
+
+class DockWinClass
+{
+
+    SaveAllWindowsFunctionName := "SaveAllWindows"
+    LoadAllWindowsFunctionName := "LoadAllWindows"
+    
+	
+	__new()
+    {  
+        this.WindowMover := new JPGIncWindowMoverClass()
+		this.desktopMapper := new DesktopMapperClass(new VirtualDesktopManagerClass())
+		this.monitorMapper := new MonitorMapperClass()  
+        return this
+    }
+    
+; goes through file
+; title of line in file is matched with open windows
+LoadAllWindows()
+{
+    SetTitleMatchMode, 3		; 2: A window's title can contain WinTitle anywhere inside it to be a match. 

```

Comment:
When you setthings like SetTitleMatchMode and DetectHiddenWindows you should save their current values and restore them after your finished with them. That way if someone makes use of this function they don't have to worry about these values getting changed